docs/api/js/types.md

```diff
@@ -1,226 +1,303 @@
-    type: 'General';
-    innerInput: BytesLike;
-}
-
-export interface EthereumSignature {
-    v: number;
-    r: BytesLike;
-    s: BytesLike;
-}
-
-export type PaymasterInput = ApprovalBasedPaymasterInput | GeneralPaymasterInput;
-
-export enum AccountAbstractionVersion {
-    None = 0,
-    Version1 = 1
-}
-
-export enum AccountNonceOrdering {
-    Sequential = 0,
-    Arbitrary = 1
-}
-
-export interface ContractAccountInfo {
-    supportedAAVersion: AccountAbstractionVersion;
-    nonceOrdering: AccountNonceOrdering;
-}
-
-export interface BatchDetails {
-    number: number;
-    timestamp: number;
-    l1TxCount: number;
-    l2TxCount: number;
-    rootHash?: string;
-    status: string;
-    commitTxHash?: string;
-    committedAt?: Date;
-    proveTxHash?: string;
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-    l1GasPrice: number;
-    l2FairGasPrice: number;
-}
-
-export interface BlockDetails {
-    number: number;
-    timestamp: number;
-    l1BatchNumber: number;
-    l1TxCount: number;
-    l2TxCount: number;
-    rootHash?: string;
-    status: string;
-    commitTxHash?: string;
-    committedAt?: Date;
-    proveTxHash?: string;
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-}
-
-export interface TransactionDetails {
-    isL1Originated: boolean;
-    status: string;
-    fee: BigNumberish;
-    initiatorAddress: Address;
-    receivedAt: Date;
-    ethCommitTxHash?: string;
-    ethProveTxHash?: string;
-    ethExecuteTxHash?: string;
-}
-```
+# Types and interfaces
+
+## `AccountAbstractionVersion`
+
+Enumerated list of account abstraction versions.
+
+- None = `0` (used for contracts that are not accounts)
+- Version1 = `1`
+
+## `AccountNonceOrdering`
+
+Enumerated list of account nonce ordering formats.
+
+- Sequential = `0`
+- Arbitrary = `1`
+
+## `Address`
+
+0x-prefixed, hex-encoded, Ethereum account address as string.
+
+## `ApprovalBasedPaymasterInput`
+
+Interface representation of approval based paymaster input containing various fields including an `ApprovalBased` type.
+
+- `type`: `'ApprovalBased'`;
+- `token`: `Address`;
+- `minimalAllowance`: `BigNumber`;
+- `innerInput`: `BytesLike`;
+
+## `BalancesMap`
+
+Type defining a map object containing accounts and their balances.
+
+- `{ [key: string]`: `BigNumber }`
+
+## `BatchDetails`
+
+Interface representation of batch information containing various optional and mandatory fields.
+
+- `number`: `number`;
+- `timestamp`: `number`;
+- `l1TxCount`: `number`;
+- `l2TxCount`: `number`;
+- `rootHash?`: `string`;
+- `status`: `string`;
+- `commitTxHash?`: `string`;
+- `committedAt?`: `Date`;
+- `proveTxHash?`: `string`;
+- `provenAt?`: `Date`;
+- `executeTxHash?`: `string`;
+- `executedAt?`: `Date`;
+- `l1GasPrice`: `number`;
+- `l2FairGasPrice`: `number`;
+
+## `Block`
+
+Interface representation of a block that extends the Ethers `providers.Block` definition with additional fields.
```

Comment:
nit: I would added a link to `providers.Block` docs here, and in similar places as well

Reply:
DONE for all